aea/cli/search.py

```diff
@@ -103,121 +103,136 @@ def _get_details_from_dir(
         results.append(details)
 
 
+def _search_items(ctx, item_type_plural):
+    registry = cast(str, ctx.config.get("registry_directory"))
+    result = []  # type: List[Dict]
+    configs = {
+        "agents": {"loader": ctx.agent_loader, "config_file": DEFAULT_AEA_CONFIG_FILE},
+        "connections": {
+            "loader": ctx.connection_loader,
+            "config_file": DEFAULT_CONNECTION_CONFIG_FILE,
+        },
+        "contracts": {
+            "loader": ctx.contract_loader,
+            "config_file": DEFAULT_CONTRACT_CONFIG_FILE,
+        },
+        "protocols": {
+            "loader": ctx.protocol_loader,
+            "config_file": DEFAULT_PROTOCOL_CONFIG_FILE,
+        },
+        "skills": {
+            "loader": ctx.skill_loader,
+            "config_file": DEFAULT_SKILL_CONFIG_FILE,
+        },
+    }
+    if item_type_plural == "agents":
+        lookup_dir = registry
+    else:
+        lookup_dir = AEA_DIR
+        _get_details_from_dir(
+            configs[item_type_plural]["loader"],
+            registry,
+            "*/{}".format(item_type_plural),
+            configs[item_type_plural]["config_file"],
+            result,
+        )
+
+    _get_details_from_dir(
+        configs[item_type_plural]["loader"],
+        lookup_dir,
```

Comment:
I would change here `lookup_dir` to registry.

Reply:
That's a bit different cause we look for items in registry and AEA_DIR, and for agents only in registry. Looks at those 2 calls.